eng/pipelines/templates/stages/archetype-java-release-patch.yml

```
@@ -0,0 +1,317 @@
+                            ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                            Target: MavenCentral
+                            BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
+                            JavaRepoRoot: $(Pipeline.Workspace)/azure-sdk-for-java
+                        - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
+                          parameters:
+                            ArtifactName: ${{parameters.ArtifactName}}-${{artifact.name}}-mavencentral-$(System.JobAttempt)
+                            ArtifactPath: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                      # After publishing to Maven, publish to the azure-sdk-for-java feed. The reason for
+                      # this is that the azure-sdk-for-java feed will have the package available immediately
+                      # whereas Maven can take several hours for the package to become available.
+                      - template: /eng/pipelines/templates/steps/java-publishing.yml
+                        parameters:
+                          ArtifactID: ${{artifact.name}}
+                          GroupID: ${{artifact.groupId}}
+                          ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                          RepositoryUrl: https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-java/maven/v1
+                          Target: JavaDevFeed
+                          BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
+                          JavaRepoRoot: $(Pipeline.Workspace)/azure-sdk-for-java
+                      - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
+                        parameters:
+                          ArtifactName: ${{parameters.ArtifactName}}-${{artifact.name}}-javadevfeed-$(System.JobAttempt)
+                          ArtifactPath: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+
+        - deployment: UpdatePackageVersion
+          displayName: "Update Package Version"
+          condition: and(succeeded(), ne(variables['Skip.UpdatePackageVersion'], 'true'))
+          environment: ${{ parameters.Environment }}
+          dependsOn: PublishPackage
+          variables:
+            - template: /eng/pipelines/templates/variables/globals.yml
+          pool:
+            name: azsdk-pool-mms-win-2019-general
+            vmImage: MMS2019
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                  - download: none
+
+                  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+                    parameters:
+                      Paths:
+                        - '**/*.xml'
+                        - '**/*.md'
+                        - '!sdk/**/test-recordings'
+                        - '!sdk/**/session-records'
```

We found that adding `/*` greatly speeds up sparse checkout in these cases:

```suggestion
                        - '!sdk/**/test-recordings/*'
                        - '!sdk/**/session-records/*'
```

---

Fixed locally, it'll be updated in the next commit.